In [9]:
import pandas as pd
from pathlib import Path

MODEL_METRICS_PATH = Path("../data/processed/model_metrics.csv")
df = pd.read_csv(MODEL_METRICS_PATH, index_col=0)

In [10]:
# look at the model metrics df
df.head()

,Accuracy Score,Precision,Recall,ROC AUC Score
Logistic Regression,0.683,0.779,0.644,0.754
Random Forest,0.709,0.730,0.804,0.762
XGBoost,0.738,0.739,0.857,0.799


## Machine Learning Conclusions

I used three machine learning models, i.e. logistic regression, random forest classifier and XGBoost to predict buyer reordering event. XGBoost achieved the strongest overall classification performance.

1.	All models achieved relatively strong accuracy scores, with XGBoost performing best at approximately 74%. However, accuracy alone is insufficient for evaluation because the dataset is moderately imbalanced, with reorder events representing roughly 59% of observations.
2.	Logistic Regression achieved the highest precision score (78%), meaning that when the model predicts a reorder event, it is correct most of the time. This suggests that Logistic Regression produces fewer false positives compared to the other models.
3.	Logistic Regression achieved lower recall (64%), indicating that it misses a larger number of actual reorder events. Random Forest improved recall substantially (80%), while XGBoost achieved the strongest performance at 86%, meaning it was most effective at identifying true reorder behavior.
4.	XGBoost achieved the highest ROC AUC score (0.80), indicating that it was the most effective model at distinguishing reorder events from non-reorder events across different decision thresholds.
